src/kubernetes_backup/backup.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python
+
+import os
+import sys
+
+import argparse
+import logging
+
+from kubernetes_backup.backup import export_and_backup
+from kubernetes_backup.config import setup_logging
+
+def parse_args():
+  parser = argparse.ArgumentParser(
+    formatter_class=argparse.ArgumentDefaultsHelpFormatter
+  )
+  parser.add_argument('--loglevel',
+    choices=['DEBUG', 'INFO', 'WARNING', 'ERROR', 'CRITICAL'],
+    default='INFO',
+    help='log level'
+  )
+  parser.add_argument('--s3', action='store_true', help='whether to backup to s3')
+  return parser.parse_args()
+
+def get_env():
+  ''' get vars from env '''
+  # set this if running locally
+  context = os.environ.get('KUBECTL_CONTEXT', '')
+
+  # set this if running inside kubernetes
+  k8s_cluster = os.environ.get('K8S_CLUSTER', '')
+
+  # S3 bucket to backup to
+  s3_bucket = os.environ.get('K8S_BACKUP_S3_BUCKET', '')
+
+  # S3 prefix to backup under
+  s3_prefix = os.environ.get('K8S_BACKUP_S3_PREFIX', 'dev/backups')
+
+  # local dir to store yamls exported from Kubernetes
+  basedir = os.environ.get('BASEDIR', '/tmp/kubernetes-backups')
```

Comment:
Minor point: I wonder if a dummy .env.example file should be checked in so that the dev knows what needs to be set. You could then use `python-dotenv` to load the vars from an `.env` file instead. Since that is a pattern we use elsewhere.